### DISCUSSION TITLE
如何添加前端路径？

### DISCUSSION THREAD

**Discussion by ventusoon** [AlistGo/alist#370] [Q&A] [answered]
使用反代之后，域名进入显示404，
反代的目录设置的云盘路径。
![微信截图_20220115125817](https://user-images.githubusercontent.com/14863512/149609617-b7c63ca6-22c9-4028-b876-daf45d3c72f0.png)
![微信截图_20220115125753](https://user-images.githubusercontent.com/14863512/149609624-57e4c6be-4553-4310-8af6-f143b0c6fc68.png)
![c4654523c2d124ee35f45e67e9f46ca](https://user-images.githubusercontent.com/14863512/149609630-5ab07e6e-064d-42a8-b6a0-282ad5b222ae.png)

**Comment by xhofe** [comment, marked as answer]
什么意思？反代到二级目录？

**Reply by ventusoon** [reply]
> 什么意思？反代到二级目录？

对的，反代到二级目录，但是alist没有找到设置二级目录的地方..........

**Reply by xhofe** [reply]
建议使用二级域名而不是二级目录，二级目录还需要配置多个目录的反代。大致有/p /public /api /d /assets 这些，具体未测试过。

**Reply by ventusoon** [reply]
> 建议使用二级域名而不是二级目录，二级目录还需要配置多个目录的反代。大致有/p /public /api /d /assets 这些，具体未测试过。

额，主要是不想解析太多二级域名，想将x-ui、alist等工具反代到同一个域名上，方便管理

**Reply by ventusoon** [reply]
还是感谢作者回复，辛苦了 @Xhofe

**Reply by fangfang333** [reply]
俺也一样，想反代到二级目录，试了多种nginx配置都失败了，nginx配置里只有  location / {  }  才能成功，用location /alist {  }怎么试都是失败。

**Reply by ventusoon** [reply]
> 俺也一样，想反代到二级目录，试了多种nginx配置都失败了，nginx配置里只有 location / { } 才能成功，用location /alist { }怎么试都是失败。

的确，只能用二级域名。。。

**Comment by yigedabuliu** [comment]
找到你 Alist 目录下 /data/config.json 中的 site_url 必须与[反向代理]后的访问地址一致，否则会生成错误的 API 路径
"site_url": "[http://127.0.0.1:5244[你的域名或ip]/alist[你的目录]]
